### PR TITLE
Module2: Webpack configuration

### PR DESCRIPTION
**Pull request by nkorczyk**
I have prepared JavaScript Development Environment:
- I have created `package.json` file.
- I have installed `React`, `Redux`, `React-Redux`, `React-Router`, `Jest`.
- I have installed and configured `webpack` & `babel`.
- I have configured tests script.
- I have set `DEV` and `PROD` build configuration.

Additionally:
- I have set up `Editorconfig`. 
- I have configured `Travis` for Continuous Integration Server.
- I have added handling CSS styles.

Please check and review.

### CODE REVIEW

File: buildScripts/srcServer.js [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+import path from 'path';
+import open from 'open';
 
-var port = 3000;
-var app = express();
+const port = 3000;
+const app = express();
 
-app.use(express.static('src'))
+// app.use(express.static('src'));
```
Comment (MichalWierzba) [inline buildScripts/srcServer.js]: Please remove commented code or serve `dist` directory to keep serving frontend app properly.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,10 @@
   "entry point": "srcServer.js",
   "main": "script.js",
   "scripts": {
-    "start": "node buildScripts/srcServer.js"
+    "start": "webpack-dev-server --hot --port 80",
+    "dev": "webpack --mode development",
+    "prod": "webpack --mode production",
```
Comment (MichalWierzba) [inline package.json]: There is nice convention to group related scripts by prefix. For example: `build:dev` and `build:prod`.

File: package.json [hunk 3]
```diff
@@ -17,8 +20,31 @@
     "url": "https://github.com/nkorczyk/ReactJSTraining/issues"
   },
   "homepage": "https://github.com/nkorczyk/ReactJSTraining#readme",
+  "dependencies": {
+    "@types/react": "16.3.11",
+    "@types/react-dom": "16.0.5",
```
Comment (MichalWierzba) [inline package.json]: Are you going to use typescript?

File: package.json [hunk 4]
```diff
@@ -17,8 +20,31 @@
...
+    "react": "16.3.2",
+    "react-dom": "16.3.2",
+    "react-redux": "^5.1.1",
+    "react-router": "4.2.0",
+    "react-router-config": "1.0.0-beta.4",
+    "react-router-dom": "4.2.2",
+    "redux": "^3.6.0",
+    "webpack": "4.6.0",
+    "webpack-cli": "2.0.14",
+    "webpack-dev-server": "3.1.3"
```
Comment (MichalWierzba) [inline package.json]: All babel and webpack-related packages can be moved to dev dependencies.

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+const path = require("path");
+const webpack = require("webpack");
+const HtmlWebpackPlugin = require("html-webpack-plugin");
+
+module.exports = function (env, options) {
+  const isProduction = options.mode === "production";
```
Comment (MichalWierzba) [inline webpack.config.js]: I'd suggest using `env` here. See https://webpack.js.org/guides/environment-variables/ for details.

File: buildScripts/srcServer.js (hunk 1 above)
Comment (nkorczyk) [reply buildScripts/srcServer.js]: Unnecessary code removed. `dist` directory is served instead of `src`.

File: package.json (hunk 2 above)
Comment (nkorczyk) [reply package.json]: Changed to `build:dev` and `build:prod`.

File: package.json (hunk 3 above)
Comment (nkorczyk) [reply package.json]: Unnecessary dependencies removed.

File: package.json (hunk 4 above)
Comment (nkorczyk) [reply package.json]: All dependencies moved to `devDependencies`.

File: webpack.config.js (hunk 5 above)
Comment (nkorczyk) [reply webpack.config.js]: Changed.

**Review by MichalWierzba — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**